File: DataBinding2WayJava/app/src/main/java/com/websarva/wings/android/databinding2wayjava/data/model/Rectangle.java
package com.websarva.wings.android.databinding2wayjava.data.model;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;

import com.websarva.wings.android.databinding2wayjava.BR;

public class Rectangle extends BaseObservable {
	private int _height = 0;
	private int _width = 0;

	public int getArea() {
		return _height * _width;
	}

	@Bindable
	public String getAreaStr() {
		return String.valueOf(getArea());
	}

	@Bindable
	public String getHeightStr() {
		return String.valueOf(_height);
	}

	@Bindable
	public String getWidthStr() {
		return String.valueOf(_width);
	}

	@Bindable
	public void setHeightStr(String heightStr) {
		if(!heightStr.equals("")) {
			int height = Integer.valueOf(heightStr);
			setHeight(height);
		}
	}

	@Bindable
	public void setWidthStr(String widthStr) {
		if(!widthStr.equals("")) {
			int width = Integer.valueOf(widthStr);
			setWidth(width);
		}
	}

	public void setHeight(int height) {
		_height = height;
		notifyPropertyChanged(BR.heightStr);
		notifyPropertyChanged(BR.areaStr);
	}

	public void setWidth(int width) {
		_width = width;
		notifyPropertyChanged(BR.widthStr);
		notifyPropertyChanged(BR.areaStr);
	}
}

File: DataBindingObservableKotlin/app/src/main/java/com/websarva/wings/android/databindingobservablekotlin/data/model/Rectangle.kt
package com.websarva.wings.android.databindingobservablekotlin.data.model

import androidx.databinding.BaseObservable
import androidx.databinding.Bindable
import com.websarva.wings.android.databindingobservablekotlin.BR

class Rectangle: BaseObservable() {
	var height: Int = 0
		set(value) {
			field = value
			notifyPropertyChanged(BR.heightStr)
			notifyPropertyChanged(BR.areaStr)
		}
	var width: Int = 0
		set(value) {
			field = value
			notifyPropertyChanged(BR.widthStr)
			notifyPropertyChanged(BR.areaStr)
		}

	fun getArea(): Int {
		return height * width
	}
	@Bindable
	fun getAreaStr(): String {
		return getArea().toString()
	}

	@Bindable
	fun getHeightStr(): String {
		return height.toString()
	}

	@Bindable
	fun getWidthStr(): String {
		return  width.toString()
	}
}
